Handwired Skeeb Keyboard

su8044 opened the pull request:
Handwired Keyboard based on the Ergodash. with an OLED similar to Kyria

## Description

5 Rows, 14 Columns Split Keyboard with thumb clusters. no rgb or led. just an oled screen with keypresses, layer, keylock and modifiers.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [  ] Core
- [  ] Bugfix
- [  ] New feature
- [  ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [  ] Documentation

## Checklist

- [  ] My code follows the code style of this project.
- [  ] My change requires a change to the documentation.
- [  ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [  ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on keyboards/handwired/myskeeb/config.h:
```diff
@@ -0,0 +1,48 @@
+
+// unknown
+
+#define CONFIG_H
```
```suggestion
```

su8044 replied:
Hi Drashna. Im not sure what does this means in github. should i change something? is #define CONFIG_H to be deleted? thanks

noroadsleft replied:
Drashna is requesting that you delete Line 46. The `#pragma once` on Line 1 means there's no reason to use this definition.

drashna replied:
Sorry for not being more verbose, but yeah, this line should be removed, completely

noroadsleft commented on keyboards/handwired/myskeeb/keymaps/default/default_e6b9980.json:
```diff
@@ -0,0 +1 @@
+{"keyboard":"ergodash/rev1","keymap":"default_e6b9980","layout":"LAYOUT","layers":[["KC_ESC","KC_1","KC_2","KC_3","KC_4","KC_5","KC_GRV","KC_MINS","KC_6","KC_7","KC_8","KC_9","KC_0","KC_BSPC","KC_CAPS","KC_Q","KC_W","KC_E","KC_R","KC_T","KC_LBRC","KC_RBRC","KC_Y","KC_U","KC_I","KC_O","KC_P","KC_EQL","KC_TAB","KC_A","KC_S","KC_D","KC_F","KC_G","KC_NO","KC_NO","KC_H","KC_J","KC_K","KC_L","KC_SCLN", ...
```
Why is there an Ergodash keymap here?

su8044 replied:
Hi. i downloaded the ergodash json file from qmk configurator. i used it as a base. can be deleted

noroadsleft replied:
You can remove this file from the pull request by entering

```
git rm --cached -- keyboards/handwired/myskeeb/keymaps/default/default_e6b9980.json
```

in the command line. The `--cached` argument means the file will remain on your computer; leave it out if you want to delete the file from your system as well.

noroadsleft commented on keyboards/handwired/myskeeb/rules.mk:
```diff
@@ -0,0 +1,33 @@
+BOOTLOADER = caterina
+#
+# Build Options
+#
```
```suggestion
#   change yes to no to disable
#
```

noroadsleft commented on keyboards/handwired/myskeeb/rules.mk:
```diff
@@ -0,0 +1,33 @@
+OLED_DRIVER_ENABLE = yes
+NO_USB_STARTUP_CHECK = yes
+TAP_DANCE_ENABLE = yes
+
```
Please replace Lines 16-33 with:

```Makefile
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
MIDI_ENABLE = no            # MIDI controls
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
UNICODE_ENABLE = no         # Unicode
SPLIT_KEYBOARD = yes        # Enables split keyboard support
OLED_DRIVER_ENABLE = yes
NO_USB_STARTUP_CHECK = yes
TAP_DANCE_ENABLE = yes
```

This removes the firmware size impact notes, aligns the comments, and conforms to QMK's AVR template for this file.

noroadsleft commented:
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

If you need any help with this just ask.

**You do not need to close this pull request at this time.** Until your pull request is merged, please continue to commit relevant changes to your `master` branch.

su8044 commented:
I have made the changes, however, at the time of making the pull request, it says there are 515 files that have changed. i did the same steps as the last time from https://docs.qmk.fm/#/getting_started_github but im not sure what happened. please if you can fix or revert the errors please let me know. i read https://docs.qmk.fm/#/newbs_git_using_your_master_branch before but i could not update my fork.  any extra guidance on what i did wrong will be very appreciated. thanks

noroadsleft commented:
> I have made the changes, however, at the time of making the pull request, it says there are 515 files that have changed. i read https://docs.qmk.fm/#/newbs_git_using_your_master_branch before but i could not update my fork. i believe now this was the issue after seeing the differences. any extra guidance on what i did wrong will be very appreciated. i will not touch my command line or git buttons until further guidance to avoid messing anything else, thanks

Ah, you have a merge commit in your history, which is why there are so many files here. Your PR has 3 commits, which we'll call `C1`, `C2`, and `C3` for this exercise. We'll call the commits on QMK's `master` branch by `M#`.

Your history looks like this:

```
  M0---C1------C2---C3
   \           /
   M1 ...... M379
```

`C1` is 7d6cd82, and was applied to `M0`. Meanwhile, QMK's `master` branch gained a bunch of commits. Then, `C2` (978fa5b) was applied, and has two parent commits: `C1` and `M379`. You can see this [here, near the top right](https://github.com/su8044/qmk_firmware/commit/978fa5bdb23da546c49c9758333e2afe50cb3270). What you should also notice is the text "Showing 2,389 changed files with 58,880 additions and 30,488 deletions." When you merge two branches together, the commit hashes change, and so even if the contents of the changes are the same, Git considers the histories to be separate. `C3` (253d905) shows 515 modified files because of the diverged histories.

What you want to do is back up the current state of your `master` branch. We'll do this by creating a new branch. We're only using it for a temporary backup, and we'll delete it later.

```
git branch master_old
```

Now you have two relevant branches: `master` and `master_old`, which are the same, but your `master` branch is still the one checked out. Now, we reset the current branch `master` so it matches QMK's `master` branch:

```
git fetch upstream master
git reset --hard FETCH_HEAD
```

The first command fetches the state of QMK's `master` branch (as long as `upstream` refers to QMK's GitHub repository). The second says "Rewrite my local Git repository (on my computer) so it's identical to the commit that was last fetched." When this completes, your local files and QMK's `master` branch and their histories are 100% identical.

Since you're only dealing with wanting to add the files relevant to one keyboard that exist in one directory, we can thankfully ignore every other directory. To pull the files for your keyboard, run:

```
git checkout 253d905 -- keyboards/handwired/myskeeb/
```

This command means "Write (or rewrite) the files in `keyboards/handwired/myskeeb/` to match those in the same directory from commit 253d905." Because your repo matches the state of QMK's `master` branch, none of the files exist, and so you end up with new files that are yet to be committed:

```
$ git status 
On branch master
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        new file:   keyboards/handwired/myskeeb/config.h
        new file:   keyboards/handwired/myskeeb/keymaps/default/keymap.c
        new file:   keyboards/handwired/myskeeb/myskeeb.c
        new file:   keyboards/handwired/myskeeb/myskeeb.h
        new file:   keyboards/handwired/myskeeb/oled.c
        new file:   keyboards/handwired/myskeeb/rules.mk
        new file:   keyboards/handwired/myskeeb/skeeb_font.c
```

At this point, run `git commit`. You'll get a new commit that only contains the addition of your new files. Now, run `git push --force-with-lease`. This tells Git to push the current state of your `master` branch (with only the one commit) to your GitHub fork, ignoring the fact that your local `master` branch isn't descended from the `master` branch on your fork. (Running `git push` alone would outright fail with an error message.) When this completes, your PR here will update with the new, clean history. At this point, you don't need the `master_old` branch, and you can delete it by running `git branch -D master_old`.

su8044 commented:
> At this point, run `git commit`. You'll get a new commit that only contains the addition of your new files. Now, run ......

Hi noroadsleft, Thanks for the amazing guide. im slowly starting to understand git. ill be playing a bit with https://learngitbranching.js.org/ to avoid these errors in the future.

after `git commit`, i get a screen saying 

```
"Please enter the commit message for your changes...."  
"your branch and origin master has 130 and 3 different commits each....."  
"use git pull to merge the remote branch into yours."
```
but in that screen, i don't know what to do next. it doesn't let me write or do anything else. i imagine at this point i should close msys and reopen it to make the last command "git push --force-with-lease"? in any case i left msys open at the screen that says to enter the commit message waiting for your response. and once again thanks for the guidance..

noroadsleft commented:
> after `git commit`, i get a screen saying
> 
> ```
> "Please enter the commit message for your changes...."  
> "your branch and origin master has 130 and 3 different commits each....."  
> "use git pull to merge the remote branch into yours."
> ```

My fault, I missed a step of sorts. You've ended up in a vim window. I should have had you pass the commit message as an argument.

1. Make sure caps lock is turned **OFF**
1. Hit `i`
1. Type `adding Handwired Skeeb Keyboard` and press `Esc`
1. Type `:wq` and hit `Enter`

From there, continue with `git push --force-with-lease`. You should be okay for the rest of the way.

su8044 commented:
every thing done with success. so now i should have my branch updated in both my hub and my pc?? should i do another step?

fauxpark commented on keyboards/handwired/myskeeb/skeeb_font.c:
```diff
@@ -0,0 +1,100 @@
+#    include <pgmspace.h>
+#else
+#    define PROGMEM
+#endif
```
Per #7745 this block can simply be:
```c
#include "progmem.h"
```

fauxpark commented on keyboards/handwired/myskeeb/rules.mk:
```diff
@@ -0,0 +1,34 @@
+# MCU name
+MCU = atmega32u4
+#
```
```suggestion

```

fauxpark commented on keyboards/handwired/myskeeb/rules.mk:
```diff
@@ -0,0 +1,34 @@
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+#
```
```suggestion

```

su8044 commented:
I did the changes Fauxpark recommended, but I haven't had the time to learn a bit more about git. 
For now, I would like to know if I should do a simple update of my branch by:
git checkout master
git fetch upstream
git pull upstream master
git push origin master

then push it with "git push"?

noroadsleft commented:
Just run `git push` here. Don't worry about resynchronizing your `master` branch until we get your pull request merged.

su8044 commented:
done. thanks

noroadsleft commented:
What's the output of `git log -5`?

su8044 commented:
https://pastebin.com/Sqzg7QvP

noroadsleft commented:
`git status -uno` ?

su8044 commented:
```
$ git status -uno
Refresh index: 100% (14150/14150), done.
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   keyboards/handwired/myskeeb/rules.mk
        modified:   keyboards/handwired/myskeeb/skeeb_font.c

no changes added to commit (use "git add" and/or "git commit -a")

```

noroadsleft commented:
> i assume i should now do:
> 
> ```
> git add keyboards/handwired/myskeeb/rules.mk
> git add keyboards/handwired/myskeeb/skeeb_font.c
> git push
> ```
> 
> ????

Correct.

su8044 commented:
done, 

git log -5: https://pastebin.com/eybaYkGU

```
$ git status -uno
On branch master
Your branch is up to date with 'origin/master'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   keyboards/handwired/myskeeb/rules.mk
        modified:   keyboards/handwired/myskeeb/skeeb_font.c

Untracked files not listed (use -u option to show untracked files)

```

noroadsleft commented:
Oops, just realized I misread your previous comment.

It seems you have the files added already (`Changes to be committed:`), but there was a step missing.

Run:

```
git commit -m "Apply suggestions from fauxpark"  # this was the missing step
git push
```

su8044 commented:
IT WORKS! looks like the changes are finally reflected on my git.

anything else i should do?

fauxpark commented on keyboards/handwired/myskeeb/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,63 @@
+
+// [Process User Input] ------------------------------------------------------//
+//bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+//}
```
```suggestion
```

fauxpark commented on keyboards/handwired/myskeeb/rules.mk:
```diff
@@ -0,0 +1,32 @@
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+# Build Options
```
```suggestion

# Build Options
```

fauxpark commented on keyboards/handwired/myskeeb/rules.mk:
```diff
@@ -0,0 +1,32 @@
+# MCU name
+MCU = atmega32u4
+# Bootloader selection
```
```suggestion

# Bootloader selection
```

su8044 commented:
done.
git push?

noroadsleft commented:
Yes. Any time you make changes, you should cycle `git add` -> `git commit` -> `git push`.

- `git add`: You're telling Git "I want to stage these changes." Staging is like telling your music app "Add this song to my current playlist, but don't start playback yet." You can use:
  - `git add -- .`: "Stage all the changes to files in my current directory."
    - Running this from `qmk_firmware/keyboards/handwired/myskeeb/` will stage any changes in that folder, but ignore changes from anything outside of that folder.
  - `git add -- file1 file2 file3`: "Stage the changes to file1, file2 and file3." Filenames are relative to your current working directory.
- `git commit`: "Bookmark the changes that are currently staged."
  - `git commit -m "[some text]"`: `-m` is the *message* argument, and what follows is the *commit message.* The commit message should be a short description of the changes in the commit being made. Then the commit message tells anyone reading `git log` what happened. **I highly recommend using this syntax.**
- `git push`: "Update a remote repository with the changes that are different between my local repository and the remote." You don't have to push after every commit, but it's a good idea until you begin to understand Git better.

su8044 commented:
```
su804@DESKTOP-SEQMPBJ MSYS ~/qmk_firmware
$ git status -uno
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   keyboards/handwired/myskeeb/keymaps/default/keymap.c
        modified:   keyboards/handwired/myskeeb/rules.mk

no changes added to commit (use "git add" and/or "git commit -a")

su804@DESKTOP-SEQMPBJ MSYS ~/qmk_firmware
$ git add keyboards/handwired/myskeeb/keymaps/default/keymap.c

su804@DESKTOP-SEQMPBJ MSYS ~/qmk_firmware
$ git add keyboards/handwired/myskeeb/rules.mk

su804@DESKTOP-SEQMPBJ MSYS ~/qmk_firmware
$ git commit -m "Apply more suggestions from fauxpark and small change to layout"
[master d5180ae09] Apply more suggestions from fauxpark and small change to layout
 2 files changed, 7 insertions(+), 13 deletions(-)

su804@DESKTOP-SEQMPBJ MSYS ~/qmk_firmware
$ git push
Username for 'https://github.com':
Password for 'https://su8044@github.com':
Enumerating objects: 17, done.
Counting objects: 100% (17/17), done.
Delta compression using up to 12 threads
Compressing objects: 100% (7/7), done.
Writing objects: 100% (9/9), 747 bytes | 41.00 KiB/s, done.
Total 9 (delta 6), reused 0 (delta 0)
remote: Resolving deltas: 100% (6/6), completed with 6 local objects.
To https://github.com/su8044/qmk_firmware.git
   65eeb4a93..d5180ae09  master -> master

su804@DESKTOP-SEQMPBJ MSYS ~/qmk_firmware
$ git status -uno
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit (use -u to show untracked files)

```
Alright. changes are done and pushed. i made a small change to the 0 layout. should not be any issues i think.

noroadsleft commented on keyboards/handwired/myskeeb/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,55 @@
+  else if (state->count == 3) {
+    tap_code16(S(KC_LBRC));
+    }
+  }
```
```suggestion
void tri_open (qk_tap_dance_state_t *state, void *user_data) {
    if (state->count == 1) {
        tap_code16(KC_LPRN);
    } else if (state->count == 2) {
        tap_code (KC_LBRC);
    } else if (state->count == 3) {
        tap_code16(S(KC_LBRC));
    }
}
```

noroadsleft commented on keyboards/handwired/myskeeb/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,55 @@
+  else if (state->count == 3) {
+    tap_code16(S(KC_RBRC));
+    }
+  }
```
```suggestion
void tri_close(qk_tap_dance_state_t *state, void *user_data) {
    if (state->count == 1) {
        tap_code16(KC_RPRN);
    } else if (state->count == 2) {
        tap_code (KC_RBRC);
    } else if (state->count == 3) {
        tap_code16(S(KC_RBRC));
    }
}
```

noroadsleft commented on keyboards/handwired/myskeeb/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,55 @@
+qk_tap_dance_action_t tap_dance_actions[] = {
+ [OP_QT] = ACTION_TAP_DANCE_FN (tri_open),
+ [CL_QT] = ACTION_TAP_DANCE_FN (tri_close),
+};
```
```suggestion
qk_tap_dance_action_t tap_dance_actions[] = {
    [OP_QT] = ACTION_TAP_DANCE_FN(tri_open),
    [CL_QT] = ACTION_TAP_DANCE_FN(tri_close),
};
```

noroadsleft commented on keyboards/handwired/myskeeb/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,55 @@
+
+	[1] = LAYOUT(KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_BSPC, KC_CAPS, KC_PDOT, KC_PPLS, KC_PMNS, KC_PAST, KC_PSLS, KC_MSEL, KC_MSTP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PWR, RESET, KC_TAB, KC_P6, KC_P7, KC_P8, KC_P9, KC_P0, KC_NO, KC_NO, KC_SLCK, MAGIC_HOST_NKRO, EEP_RST, KC_TRNS, KC_TRNS, KC_TRNS, KC_LSFT, KC_P1, KC_P2, KC_P3, KC_P4, KC ...
+	[2] = LAYOUT(KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_7, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_LCTL, KC_A, KC_S, KC_D, KC_F, KC_G, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_SPC, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, K ...
+	[3] = LAYOUT(KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_TAB, KC_A, KC_W, KC_E, KC_R, KC_T, KC_H, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_LSFT, KC_Z, KC_S, KC_D, KC_F, KC_G, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_LCTL, KC_NO, KC_X, KC_C, KC_V, KC_B, KC_M, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC ...
```
```suggestion
    [0] = LAYOUT(
        KC_ESC,  KC_1,    KC_2,    KC_3,   KC_4,          KC_5,   KC_EQL,             KC_MINS,    KC_6,   KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
        KC_TAB,  KC_Q,    KC_W,    KC_E,   KC_R,          KC_T,   TD(OP_QT),          TD(CL_QT),  KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_BSLS,
        KC_CAPS, KC_A,    KC_S,    KC_D,   KC_F,          KC_G,   KC_NO,              KC_NO,      KC_H,   KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
        KC_LSFT, KC_Z,    KC_X,    KC_C,   KC_V,          KC_B,   KC_CALC,            KC_PSCR,    KC_N,   KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,
        KC_LCTL, KC_LGUI, KC_LALT, KC_DEL, LT(1,KC_BSPC), KC_SPC, KC_ENT,             KC_ENT,     KC_SPC, KC_TILD, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
    ),
    [1] = LAYOUT(
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,            KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_BSPC,
        KC_CAPS, KC_PDOT, KC_PPLS, KC_PMNS, KC_PAST, KC_PSLS, KC_MSEL,          KC_MSTP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PWR,  RESET,
        KC_TAB,  KC_P6,   KC_P7,   KC_P8,   KC_P9,   KC_P0,   KC_NO,            KC_NO,   KC_SLCK, NK_ON,   EEP_RST, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_LSFT, KC_P1,   KC_P2,   KC_P3,   KC_P4,   KC_P5,   KC_MUTE,          KC_MPLY, KC_NLCK, NK_OFF,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
        KC_LCTL, KC_LGUI, KC_LALT, KC_BSPC, KC_TRNS, KC_VOLD, KC_VOLU,          KC_MPRV, KC_MNXT, TO(2),   KC_HOME, KC_END,  KC_PGUP, KC_PGDN
    ),
    [2] = LAYOUT(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,             KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_7,             KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
        KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_NO,            KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_SPC,           KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
        KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_SPC,  KC_SPC,  KC_SPC,           KC_NO,   KC_NO,   TO(3),   KC_NO,   KC_NO,   KC_LALT, KC_TAB
    ),
    [3] = LAYOUT(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,             KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
        KC_TAB,  KC_A,    KC_W,    KC_E,    KC_R,    KC_T,    KC_H,             KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
        KC_LSFT, KC_Z,    KC_S,    KC_D,    KC_F,    KC_G,    KC_NO,            KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
        KC_LCTL, KC_NO,   KC_X,    KC_C,    KC_V,    KC_B,    KC_M,             KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
        KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_SPC,  KC_SPC,  KC_Q,             KC_NO,   KC_NO,   TO(0),   KC_NO,   KC_NO,   KC_LALT, KC_TAB
    )
```

noroadsleft commented:
Congratulations on your first pull request!

Now that your pull request is merged, you should resynchronize your `master` branch so it's easier to keep your QMK Firmware updated. [The instructions here](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch "Fixing Your Branch - QMK Firmware Docs") should get you sorted. If you need any help just ask!

su8044 commented:
Excellent! I'll be creating my next keyboard in a few months based on the same design. I have learned a lot and found a new hobby.  thank you guys and gals for all your effort and your patience, amazing work and effort on your behalf!
